flake.nix

```diff
@@ -0,0 +1,155 @@
+        #       };
+        #     };
+        #   };
+        # };
+        devShells.default =
+          let
+            stableToolchain = pkgs.rust-bin.stable.latest.minimal.override {
+              extensions = [ "rustfmt" "clippy" "llvm-tools-preview" "rust-src" ];
+            };
+            # nixWithFlakes allows pre v2.4 nix installations to use
+            # flake commands (like `nix flake update`)
+            nixWithFlakes = pkgs.writeShellScriptBin "nix" ''
+              exec ${pkgs.nixFlakes}/bin/nix --experimental-features "nix-command flakes" "$@"
+            '';
+          in
+          mkShell
+            {
+              buildInputs = [
+                # Rust dependencies
+                pkg-config
+                openssl
+                curl
+                protobuf # to compile libp2p-autonat
+                stableToolchain
+
+                # Rust tools
+                cargo-edit
+                cargo-watch
+                cargo-sort
+
+                # Tools
+                nixWithFlakes
+                entr
+
+                # Figures
+                graphviz
+                plantuml
+                coreutils
+
+                # cdk-validium-node
+                go
+                docker-compose
+                postgresql_15
+              ] ++ lib.optionals stdenv.isDarwin[darwin.apple_sdk.frameworks.SystemConfiguration];
+              shellHook = ''
+                # Prevent cargo aliases from using programs in `~/.cargo` to avoid conflicts
+                # with rustup installations.
+                export CARGO_HOME=$HOME/.cargo-nix
+              ''
+              # + self.checks.${system}.pre-commit-check.shellHook
+              ;
+              RUST_SRC_PATH = "${stableToolchain}/lib/rustlib/src/rust/library";
+              RUST_BACKTRACE = 1;
+              RUST_LOG = "info,libp2p=off";
+            };
+        devShells.staticShell =
+          let
+            muslPkgs = import nixpkgs {
+              localSystem = system;
+              crossSystem = { config = "${arch}-unknown-${os}-musl"; };
+            };
+            stableMuslRustToolchain =
+              pkgs.rust-bin.stable.latest.minimal.override {
+                extensions = [ "rustfmt" "clippy" "llvm-tools-preview" "rust-src" ];
+                targets = [ "${arch}-unknown-${os}-musl" ];
+            };
+            opensslMusl = muslPkgs.openssl.override { static = true; };
+            curlMusl = (muslPkgs.pkgsStatic.curl.override {
+              http2Support = false;
+              libssh2 = muslPkgs.pkgsStatic.libssh2.dev;
+              zstdSupport = false;
+              idnSupport = false;
+            }).overrideAttrs (oldAttrs:
+              let confFlags = oldAttrs.configureFlags;
+              in {
+                configureFlags = (muslPkgs.lib.take 13 confFlags)
+                  ++ (muslPkgs.lib.drop 14 confFlags)
+                  ++ [ (muslPkgs.lib.withFeature true "libssh2") ];
```

Comment:
reminds me why do we need these openssl and libssh for? did you try removing them?

Reply:
don't think they are used (see below)